### PR TITLE
feat: Paginated Attachments

### PR DESCRIPTION
**Pull request by strongishllama**
- Removed the usage of deprecated methods in DbPDO.
- Added versions to @deprecated tags on the comments of deprecated methods.
- Added paginatedList method to the Html class.
- Replaced magic methods with getInstance() singletons.
- Improved some comments.
- Renamed some variables to camel case.
- Added $page and  $page_size parameters to FileService::getAttachments() method.
- Added attachment_item partial.
- Refactored listattachments partial to support pagination.
- Improved listattachment page load time by loading images asynchronously.
- Removed some commented code and whitespace.
- Replace array() usage with [] shortand.
- Move some long lines onto multiple to make it easier to read
- Added ability to skip prepend of prefix on attachment (@adam-buckley)
- Added some code documentation and CS fixes (@adam-buckley)

refs: 9261

### CODE REVIEW

File: system/html.php [hunk 1]
```diff
@@ -106,7 +107,7 @@ public static function dataTable($data = array())
      * @param mixed $width
      * @return string
      */
-    public static function chart($id = "chartjs", $type = "line", $data = array(), $options = array(), $height = null, $width = null, $class = null)
+    public static function chart($id = "chartjs", $type = "line", $data = [], $options = [], $height = null, $width = null, $class = null)
```
Comment (twje) [inline system/html.php:110]: Great to see the use of short-hand syntax, improves readability

**Comment by DerekCrannaford** [comment]
Looking good. I'am aware of some custom modules with their own create/clone attachment methods = will always be risky when the base properties are changed (ie: will a custom 'constructor' keep up?) Your handling of "$skip_path_prefix" looks safe in this regard.
Is there anything other than formatting changes in web?

**Comment by strongishllama** [comment]
> Looking good. I'am aware of some custom modules with their own create/clone attachment methods = will always be risky when the base properties are changed (ie: will a custom 'constructor' keep up?) Your handling of "$skip_path_prefix" looks safe in this regard.
> Is there anything other than formatting changes in web?

The web changes are all just formatting.

**Comment by DerekCrannaford** [comment]
14: listattachments.tpl: "/cpu-upload/uploadedit/$object->id#attachments"
Should CPU action be bakied into core like this? Can it be avoided, maybe by a config entry/override coming back from CPU module?

**Review by DerekCrannaford — APPROVED** [review]
Approved per response to comments.

**Pull request merged**